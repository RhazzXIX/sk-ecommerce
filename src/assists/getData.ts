export default async function getData() {
  const fetchedData = await fetch("https://dummyjson.com/products?limit=0")
    .then((res) => res.json())
    .catch((e) => {
      console.error(e);
      throw new Error("Failed to load data");
    });
  return fetchedData.products;
}
