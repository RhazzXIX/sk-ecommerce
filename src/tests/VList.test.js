import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import VList from "../app/list/page";
import getData from "../assists/getData";

jest.mock("../assists/getData");

getData.mockResolvedValue([
  {
    brand: "Apple",
    category: "smartphones",
    description: "An apple mobile which is nothing like apple",
    discountPercentage: 12.96,
    id: 1,
    price: 549,
    rating: 4.69,
    stock: 94,
    title: "iPhone 9",
  },
  {
    id: 100,
    title: "Crystal chandelier maria theresa for 12 light",
    description: "Crystal chandelier maria theresa for 12 light",
    price: 47,
    discountPercentage: 16,
    rating: 4.74,
    stock: 133,
    brand: "YIOSI",
    category: "lighting",
  },
]);

describe("List component", () => {
  it("Renders on screen", async() => {
    const { container } = render(<VList />);
    await screen.findByText('iPhone 9')
    expect(container).toMatchSnapshot();
  });
});
