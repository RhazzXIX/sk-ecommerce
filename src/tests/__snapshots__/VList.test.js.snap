// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component Renders on screen 1`] = `
<div>
  <main>
    <section>
      <section>
        <h1>
          Categories
        </h1>
      </section>
    </section>
    <section>
      <header />
      <div
        style="position: relative; height: 800px; width: 100%; overflow: auto; will-change: transform; direction: ltr;"
      >
        <div
          style="height: 100px; width: 100%;"
        >
          <div
            style="position: absolute; left: 0px; top: 0px; height: 50px; width: 100%;"
          >
            iPhone 9
          </div>
          <div
            style="position: absolute; left: 0px; top: 50px; height: 50px; width: 100%;"
          >
            Crystal chandelier maria theresa for 12 light
          </div>
        </div>
      </div>
    </section>
  </main>
</div>
`;
